src/NuGet.Core/NuGet.Protocol/PublicAPI/net472/PublicAPI.Unshipped.txt

```diff
@@ -1 +1,30 @@
 #nullable enable
+NuGet.Protocol.ClientVersions
+NuGet.Protocol.RegistrationsBaseUrlTypes
+~static readonly NuGet.Protocol.ClientVersions.Version200 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version300 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version300beta -> string
+~static readonly NuGet.Protocol.ClientVersions.Version300rc -> string
+~static readonly NuGet.Protocol.ClientVersions.Version340 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version360 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version470 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version490 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version500 -> string
+~static readonly NuGet.Protocol.ClientVersions.Version510 -> string
+~static readonly NuGet.Protocol.ClientVersions.Versioned -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version200 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version300 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version300beta -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version340 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version470 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version490 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version500 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Version510 -> string
+*REMOVED*~static readonly NuGet.Protocol.ServiceTypes.Versioned -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrl -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrlVersion300beta -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrlVersion300rc -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrlVersion340 -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrlVersion360 -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrlVersioned -> string
+~static readonly NuGet.Protocol.RegistrationsBaseUrlTypes.RegistrationsBaseUrls -> string[]
```

Comment:
I wrote some docs on the NuGet SDK and API changes a while ago: https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#changes-to-apis

Basically, are these types useful to customers who use NuGet.Protocol in their own apps? I don't think they are, as I'd expect developers to use `SourceRepository.GetResourceAsync<T>`, and all these constants are just an internal implementation detail.

Unfortunately people on the NuGet team years ago weren't mindful about what they made public vs internal, but this has bitten us, making it harder to make perf improvements, and sometimes bug fixes, which is why we're much more deliberate about public API changes now. As much as I'd love to hide a bunch of public APIs that have already shipped, as mentioned, we've had problems in the past with people using them, even though I believe there were better ways of doing what they wanted to achieve, but in any case we try not to break people. Therefore, we don't want to add new APIs that are not useful and therefore add technical debt.

Reply:
Okay, I have pushed some updates and paired down the code. In this go around I added the missing versions alongside the existing ones in the ServiceTypes file. This does create new entries in the public api, but follows the pattern of exposing the version strings that the class was already doing.